app/controllers/application_controller.rb

```diff
@@ -276,6 +280,16 @@ def render_record_not_found e
     }, status: :not_found
   end
 
+  def bad_secret_encoding e
+    logger.debug "#{e}\n#{e.backtrace.join "\n"}"
+    render json: {
+      error: {
+        code: "bad_secret_encoding",
+        message: e.message
+      }
+    }, status: :internal_server_error
```

Comment:
is this a server error or a bad request error? not sure what causes this

Reply:
It is caused by a binary secret stored without specifying the mime type. This causes the server to fail to encode it into a string to be included in a JSON object. This is why I considered it to be a server error not a request error, because there is nothing technically wrong with the request. This wouldn't be an issue if we chose to encode all secrets returned as base64 strings as [suggested by](https://github.com/cyberark/conjur/pull/1989#discussion_r555913613) @doodlesbykumbi

Reply:
thanks for the exaplanation @telday ! i agree that it should be a server error now that i understand it.

It seems that the client error was when they created the secret. am i correct? if so, can we fail such a request?

Reply:
(it can definitely be out of scope of this issue, just want to understand it)

Reply:
ok i read now your other comment. It seems that the secret itself is ok but is not only when retrieved in a batch request. Is that correct?

Reply:
Yeah, it is an issue with rendering a binary secret into the JSON document that we return in the batch endpoint. I think it would be possible to stop a user from creating a binary secret on a variable that does not have the correct mime_type annotation set, if that is something we want. I'm not totally sure if that would be useful though (at least for this endpoint based on the new implementation).